### PR TITLE
Support optional filtering functions in the image extractor.

### PR DESCRIPTION
**Pull request by JonathanSun**

### CODE REVIEW

File: pkg/find/images.go [hunk 1]
```diff
@@ -65,12 +71,18 @@ func (b *ImageFinder) AllContainerImages() []*ContainerImage {
 
 // ContainerImages returns all the images from a single Kubernetes object.
 func (b *ImageFinder) ContainerImages(key bundle.ComponentReference, st *unstructured.Unstructured) []*ContainerImage {
+	return b.FilteredContainerImages(nil, key, st)
+}
+
+// FilteredContainerImages returns all the images from a single Kubernetes object,
+// filtered by the specified function.
+func (b *ImageFinder) FilteredContainerImages(filter func(fieldName string, parentFieldName string, img string) bool, key bundle.ComponentReference, st *unstructured.Unstructured) []*ContainerImage {
```
Comment (Kashomon) [inline pkg/find/images.go:79]: What's the intent of the filter function? 

Maybe we can make a separate type and document it?

Comment (JonathanSun) [reply pkg/find/images.go]: Done.

**Comment by JonathanSun** [comment]
PTAL

**Pull request merged**